llama-index-core/llama_index/core/workflow/handler.py

```diff
@@ -73,3 +76,17 @@ async def run_step(self) -> Optional[Any]:
             raise ValueError("Context is not set!")
 
         return retval
+
+
+def get_state_dict(handler: WorkflowHandler) -> Dict[str, Any]:
+    if not handler.ctx:
+        raise ValueError("Context is not set!")
+
+    return handler.ctx.to_dict()
+
+
+def get_handler_from_state_dict(
+    workflow: "Workflow", data: Dict[str, Any]
+) -> WorkflowHandler:
+    ctx = Context.from_dict(workflow, data, serializer=workflow._serializer)
+    return WorkflowHandler(ctx=ctx)
```

Comment:
I don't think we need these, they add an indirection level for no real gain

Reply:
thats fair, I mostly just wanted a way for the user to serialize this without needing to know what a handler is lol

Reply:
But will remove in favour of exposing machinery for now